grant/grant.go

```diff
@@ -330,17 +344,11 @@ type RedeemGrantsRequest struct {
 // Returns transaction info for grant fufillment
 func (req *RedeemGrantsRequest) VerifyAndConsume(ctx context.Context) (*wallet.TransactionInfo, error) {
 	log := lg.Log(ctx)
-
-	// 1. Check grant signatures and decode
-	grants := make([]Grant, 0, len(req.Grants))
-	for _, grantJWS := range req.Grants {
-		grant, err := FromCompactJWS(grantJWS)
-		if err != nil {
-			return nil, err
-		}
-		grants = append(grants, *grant)
+	// 1. Check grant signatures and decode, filter unredeemed
+	grants, err := req.GetRedeemed(ctx)
```

Comment:
I don't think we should change VerifyAndConsume, maybe instead we call GetRedeemed in `controllers/grants.go` so we can easily return a list of already redeemed grants to the client?